cpp_faster_fifo/tests/comparison_tests.py

```diff
@@ -0,0 +1,173 @@
+import multiprocessing
+from unittest import TestCase
+from faster_fifo import Queue
+from time import time
+from queue import Full, Empty
+import logging
+
+ch = logging.StreamHandler()
+ch.setLevel(logging.DEBUG)
+
+log = logging.getLogger('rl')
+log.setLevel(logging.DEBUG)
+log.handlers = []  # No duplicated handlers
+log.propagate = False  # workaround for duplicated logs in ipython
+log.addHandler(ch)
+
+MSG_SIZE = 5
+
+
+def make_msg(msg_idx):
+    return (msg_idx,) * MSG_SIZE
+
+
+def produce_msgs(q, p_idx, num_messages):
+    i = 0
+    while i < num_messages:
+        try:
+            q.put(make_msg(i), timeout=0.01)
+            if i % 100000 == 0:
+                log.info('Produce: %d %d', i, p_idx)
+            i += 1
+        except Full:
+            # time.sleep(0.001)
+            pass
+        except Exception as exc:
+            log.exception(exc)
+    log.info('Done producing! %d', p_idx)
+
+
+def consume_msgs(q, queue_type, p_idx, consume_many=None, total_num_messages=int(1e9)):
+    messages_received = 0
+    while True:
+        try:
+            msgs = q.get_many(timeout=0.01, max_messages_to_get=consume_many) if queue_type == 'ff' else q.get()
+            if queue_type == 'mp' and msgs is None:
```

Comment:
I agree that sending a None message is the easiest way to terminate the consumer after all messages are 100% processed. I think we should use this for both our and mp queues.
It works perfectly for 1-message-at-a-time case, and for consume_many case you should just send num_consumers * consume_many messages to the queue, this should be enough to terminate all processes.

Alternatively you can use get() with small timeout, and then use something like multiprocessing.Event to signal that consumer processes should be terminated.

Bottom line: let's use a single termination mechanism for both cases. None msg is fine. You can write something like:
```
if any(m is None for m in msgs):
    break
```
this should work for all cases.

Reply:
Done